File: Lab6/Question6/src/Circle.java
public class Circle extends Shape{
    private double radius;

    public Circle(double x, double y, double radius) {
        super(x, y);
        this.radius = radius;
    }

    @Override
    public double area() {
        return Math.PI * radius * radius;
    }

    public static void testCircle() {
        Circle circle = new Circle(2, 4, 3);
        //Test attributes of circle
        System.out.println(circle.getX() == 2);
        System.out.println(circle.getY() == 4);
        System.out.println(circle.area() == 9 * Math.PI);
    }
}

File: Lab6/Circle.kt
import kotlin.math.PI

class Circle : Shape {
    private val radius: Double

    constructor(x: Double, y: Double, radius: Double) : super(x, y) {
        this.radius = radius
    }

    override fun area(): Double {
        return PI * radius * radius
    }

    override fun toString(): String {
        return "${this.javaClass.name} has area ${area()}"
    }

    companion object {
        fun testCircle() {
            val circle = Circle(2.0, 4.0, 3.0)
            println(circle.getX() == 2.0)
            println(circle.getY() == 4.0)
            println(circle.area() == 9 * PI)
        }
    }
}